### ISSUE TITLE
Allow the *.typ file to be called something other than main.typ

### ISSUE THREAD

**Issue by Enivex** [Cubxity/typstudio#25]
This caused massive headaches for me ;)

At the very least, this limitation should be written clearly somewhere.

**Comment by Cubxity** [comment]
This is possible, but currently undocumented. It is possible to set the path to the main file in `.typstudio/config.json`.

**Comment by Enivex** [comment]
> This is possible, but currently undocumented. It is possible to set the path to the main file in `.typstudio/config.json`.

Do I have to create the file myself?

Anyway, ideally it would detect what the main file is. Especially if there's only a single .typ file in the folder.

**Comment by Cubxity** [comment]
Yes, you have to manually create the file for the time being. There will be a user-friendly way to set this in the GUI in the future.

**Comment by caoimhebyrne** [comment]
> There will be a user-friendly way to set this in the GUI in the future.

We should add a warning that appear when trying to compile a file without a `main.typ`, and it will then ask "Do you want to mark the current file as the main typst file?", if the user selects yes, we will write the `config.json` for them/make changes to it.

We could also:
- Add a context menu action to files that allows the user to mark it as the main typst file
- Add a project creation wizard which will create a `main.typ` file automatically with some boilerplate within it

What do you think of these suggestions @Cubxity?
